### PR TITLE
refactor: override HasStyle implementation in ConfirmDialog

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Same as #4549 but for `ConfirmDialog`.

Based on #4573

## Type of change

- Refactor

### CODE REVIEW

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java [hunk 1]
```diff
@@ -21,8 +21,7 @@ public StylingPage() {
         addClassNameFoo.setId("add-foo");
 
         Button setClassNameBar = new Button("Set class bar", e -> {
-            dialog.setClassName("foo bar");
-            dialog.getClassNames().set("foo", false);
+            dialog.setClassName("bar");
```
Comment (web-padawan) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java:24]: Looks like the case in this IT isn't covered by the logic of `InternalOverlayClassListProxy` 😕 
However, this approach looks like an edge case to me, so we can probably accept it.

Comment (tomivirkki) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java:24]: This could be fixed by implementing `setClassName` to support multiple class names split by whitespace. Something like:

```java
public void setClassName(String className) {
    getClassNames().clear();
    if (className != null) {
        getClassNames().addAll(Arrays.asList(className.split(" ")));
    }
}
```

Comment (web-padawan) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java]: Thanks, done.

Comment (tomivirkki) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java]: Makes sense to add the same change to `Dialog` also.

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java [hunk 2]
```diff
@@ -29,7 +28,9 @@
 import com.vaadin.flow.component.UI;
 import com.vaadin.flow.component.dependency.JsModule;
 import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.component.shared.InternalOverlayClassListProxy;
```
Comment (knoobie) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:31]: Would it make sense to use the same package semantic flow is using? "shared.internal.ClassName" so that don't even see this class / interace when they navigate to the package? This could also allow to drop internal from the name

Comment (web-padawan) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:31]: Thanks, makes sense 👍  I will do that in a separate PR after this one lands.

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java (hunk 1 above)
Comment (web-padawan) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/StylingPage.java]: Thanks, done.

**Review by tomivirkki — APPROVED** [review]

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java (hunk 2 above)
Comment (web-padawan) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:31]: Created #4581 to move the class as you suggested.
